[debugger] Removing unhandled_exception which was used for android.

thaystg opened the pull request:
Trying to remove the usage of unhandled_exception function to make debugger handle with an exception when running on android.

The biggest difference is on xamarin-android side.
Where I removed the try catch added when generating dynamic methods while debugging and let the debugger handle the caught or uncaught exceptions.

Relates to xamarin/xamarin-android#4927

CoffeeFlux commented:
This was the PR that previously broke things, right? So you've changed X-A so that it's now fine to do?

thaystg commented:
Yes, this is the PR that broken things, but the part that break things I removed, and changed on xamarin-android side.
This is what I removed: https://github.com/thaystg/mono/blob/75ccd3e10f5605b70b13f82020229938523eb8a8/mono/mini/mini-exceptions.c#L2372

CoffeeFlux commented:
@monojenkins build failed

thaystg commented:
We should wait xamarin/xamarin-android#4927 to merge.

akoeplinger commented:
@thaystg is the xamarin-android PR still in progress or are we using a different solution?

thaystg commented:
Still in progres..

thaystg commented:
Closing for now. When I have time to look at this again I will open another PR.
